acid/features/history/model.py

```diff
@@ -76,9 +76,13 @@ def get_for_pipeline(cls, pipeline):
             len(select(b for b in bs.builds)) > 0).sort_by(desc(cls.id))
 
     @classmethod
-    def get_filtered(cls, pipeline, branch='', build=''):
-        if branch not in cls.get_branches():
-            branch = ''
+    def get_filtered(cls, pipeline, branches, build=''):
+        for index, branch in enumerate(branches):
+            if branch not in cls.get_branches():
```

Comment:
you are calling `cls.get_branch()` method in loop. You should call it once before loop and use its result instead.
Instead of removing branches by index, you can use *comprehensive list* and make it more pythonic.

```python
all_branches = cls.get_branches()
branches = [x for x in branches if x in all_branches]
```